lib/ui/widgets/schedule_calendar_tile.dart

```diff
@@ -44,8 +44,15 @@ class _ScheduleCalendarTileState extends State<ScheduleCalendarTile> {
     final teacherName = courseInfos[3];
     final startTime =
         "${widget.start.hour}:${widget.start.minute.toString().padLeft(2, '0')}";
-    final endTime =
-        "${widget.end.hour}:${widget.end.add(const Duration(minutes: 1)).minute.toString().padLeft(2, '0')}";
+    var endTime = "";
+    if (widget.end.minute == 59) {
+      // If the minutes are 59, add an hour and set minutes to 00
+      endTime = "${widget.end.add(const Duration(hours: 1)).hour}:00";
+    } else {
+      // Otherwise, just add a minute to the current end time
+      endTime =
+          "${widget.end.hour}:${widget.end.add(const Duration(minutes: 1)).minute.toString().padLeft(2, '0')}";
+    }
```

Comment:
Quelque chose comme ça serait pt plus clean pour formatter les dates t'en penses quoi?

```suggestion
    final DateFormat formatter = DateFormat('H:m');
    final String formatted = formatter.format(widget.end.add(const Duration(minutes: 1)));
    var endTime = formatted;
```

Reply:
J'ai changé le formatting de endTime pour quelque chose qui ressemble à ça, laisse moi savoir ce que t'en penses :)